### ISSUE TITLE
Idea: recognise default values

### ISSUE THREAD

**Issue by jonom** [littlegiant/silverstripe-seo-editor#17]
Hey @stevie-mayhew, what's up 😀

Just trying this module, it looks handy. I was thinking though, that in real world use (for me at least) all my pages have a meta title whether I provide a custom one or not, because I'll generate one if not provided. I'll sometimes do the same for Meta Descriptions too. So I would like error messages to reflect the actual tags that are output, and I would like to see the default content when a field is blank, to know whether it's worth overriding it with a custom value.

What do you think about providing a method like this to determine a default value:

```php
// Provide a default meta title if no custom one is set
public function defaultMetaTitle() {
    return $this->Title . ' → ' . SiteConfig::current_site_config()->Title;
}
```

This could be overridden in the `Page` class to let developers define their own default. Then you could set a placeholder on form fields so that any blank fields display their default values, and show error messages based on the final resolved value rather than the custom value only.

To explain my approach further - to bring this all together I would then put something like this in my `Page` class:

```php
// Resolve the actual meta title content that will be output
public function resolvedMetaTitle() {
    return $this->MetaTitle ?: $this->defaultMetaTitle();
}
```

And I'd put this in my `PageController`. (Putting it on the model would cause the resolved value to be populated on MetaTitle form fields)

```php
// Resolve a meta title for this page
public function getMetaTitle() {
    return $this->resolvedMetaTitle();
}
```

**Comment by stevie-mayhew** [comment]
Seems like a good idea as long as its configurable to turn on and off - we like to show clients where they haven't done anything beyond the default. Or we could somehow mark it as a "default value"?

**Comment by jonom** [comment]
Yeah the way I'm using it locally is I set a placeholder value on the field with the default value. So the user can see whether or not they've provided a custom value, and what the default value is if they haven't provided a custom one. That's the idea at least, seems for now that placeholder values and field values look identical in SS4 😅 so I might have to open an issue about that.

**Comment by jonom** [comment]
Relevant: https://github.com/silverstripe/silverstripe-cms/pull/2215

**Comment by jonom** [comment]
Little update, that PR has been merged into 4.4.x-dev, so should open up some new possibilities for reading and writing meta tags
